cloudinit/controlplane_init.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloudinit
+
+import (
+	"github.com/pkg/errors"
+)
+
+const (
+	controlPlaneCloudInit = `{{.Header}}
+{{template "files" .WriteFiles}}
+-   path: /tmp/kubeadm.yaml
+    owner: root:root
+    permissions: '0640'
+    content: |
+      ---
+{{.ClusterConfiguration | Indent 6}}
+      ---
+{{.InitConfiguration | Indent 6}}
+kubeadm:
+  operation: init
+  config: /tmp/kubeadm.yaml
+`
+)
+
+// ControlPlaneInput defines the context to generate a controlplane instance user data.
+type ControlPlaneInput struct {
+	baseUserData
+	Certificates
+
+	AdditionalFiles      []Files
```

Comment:
perhaps we should add an `AdditionalCommands` and then requirements like the aws bastion can be easily accommodated?